src/base/net/downloadmanager.h

```diff
@@ -67,10 +109,28 @@ namespace Net
         explicit DownloadManager(QObject *parent = nullptr);
 
         void applyProxySettings();
+        void handleReplyFinished(QNetworkReply *reply);
 
         static DownloadManager *m_instance;
         QNetworkAccessManager m_networkManager;
+
+
+        QSet<ServiceID> m_sequentialServices;
+        QSet<ServiceID> m_busyServices;
+
+        struct Job
+        {
+            QNetworkRequest request;
+            DownloadHandler *handler;
+
+            bool operator==(const Job &other);
```

Comment:
should this be a free function?

Reply:
Unfortunately I couldn't make it a free function since Job is private. Even if I declare it as friend. Maybe I make it in a wrong way...

Reply:
There should be no problem with that. Simply add a friend declaration to the DownloadManager class and then declare the operator inside the Net namespace.

Reply:
Hell, I was worried about doing it because of the difference in compilers.  In the end, I did not understand one thing:
```c++
namespace Net
{
    class DownloadManager
    {
        struct Job {/*...*/};

        friend bool operator==(const DownloadManager::Job &lhs, const DownloadManager::Job &rhs);
    };

    bool operator==(const DownloadManager::Job &lhs, const DownloadManager::Job &rhs);
}

// this doesn't compile with MSVC (cannot access private struct Job)
bool Net::operator==(const DownloadManager::Job &lhs, const DownloadManager::Job &rhs)
{
    //...
}

// but it compiles this way. What's it?
namespace Net
{
    bool operator==(const DownloadManager::Job &lhs, const DownloadManager::Job &rhs)
    {
        //...
    }
}
```

Reply:
Anyway I rewrote code so it doesn't needed anymore.